### PR TITLE
[DDAN-82] exs edit

### PR DESCRIPTION
**Pull request by ptaejoon**
### PURPOSE OF PR
* Simple summary of this PR
add new feature : exs edit

### What type of PR is it?
* fix / feature / improvement /  hot-fix / refactoring
feature

### JIRA ISSUE
* https://ddangcoonfire.atlassian.net/browse/DDAN-82

### HAVE YOU CHECKED ...
* [x] ADD JIRA ID ON PR TITLE (ex. [DDAN-1234] PR TITLE)
* [x] pre-commit
* [ ] Branch Format ( type/issue-NUM ) (ex. refactoring/DDAN-1234 )

### Explanation
( Adding screenshot would be great help )
New command `exs edit` added.
For task running, kill running task and add new command to queued_tasks 
For queued task, replace its command.

### CODE REVIEW

File: experiment_scheduler/master/master.py [hunk 1]
```diff
@@ -292,6 +298,40 @@ def _wrap_by_task_status(self, task_id, status):
             status=status,
         )
 
+    def edit_task(self, request, context):
+        task_id = request.task_id
+        cmd = request.cmd
+        if task_id in self.queued_tasks.keys():
+            self.queued_tasks[task_id].command = cmd
+            response = MasterResponse(
+                experiment_id="0", response=MasterResponse.ResponseStatus.SUCCESS
```
Comment (ptaejoon) [inline experiment_scheduler/master/master.py:307]: Currently MasterResponse returns "0" since edit response does not need experiment id.

Comment (ptaejoon) [inline experiment_scheduler/master/master.py:307]: if edit target task is in queued_tasks, change it's command only

File: experiment_scheduler/master/master.py [hunk 2]
```diff
@@ -292,6 +298,40 @@ def _wrap_by_task_status(self, task_id, status):
...
+
+            task_name = task_id.split("-")[0]
+            command = cmd
+            task_env = dict()
+            self.queued_tasks[task_id] = MasterTaskStatement(
+                name=task_name, command=command, task_env=task_env
+            )
+            response = MasterResponse(
+                experiment_id="0",
+                response=MasterResponse.ResponseStatus.SUCCESS,  # pylint: disable=E1101
```
Comment (ptaejoon) [inline experiment_scheduler/master/master.py:324]: if edit target task is running, kill its task and put changed task to queued_task.
However, task_env does not served properly. To make queued_task instance with running_tasks, running task must have queued_task's member values.

File: experiment_scheduler/submitter/edit.py [hunk 3]
```diff
@@ -1,2 +1,50 @@
...
+
+def parse_args():
+    """
+    parse only tid, cmd option.
+    :return: parsed_arguments
+    """
+    parser = argparse.ArgumentParser(description="Execute exeperiments.")
+    parser.add_argument("-t", "--tid")
+    parser.add_argument("-c", "--cmd")
+    return parser.parse_args()
```
Comment (ptaejoon) [inline experiment_scheduler/submitter/edit.py:21]: edit command :
```bash
exs edit -t {task_id} -c {command_to_change}
```
ex :
```bash
exs edit -t task1-asfkldanfk -c "python sample.py -a 10"
```

**Review by sinunu — COMMENTED** [review]
Thanks for your work! It's quite nice feature.
Could you update README file and add unit tests?

File: experiment_scheduler/submitter/edit.py
[the remark this answers is not in the document]
Comment (ptaejoon) [reply experiment_scheduler/submitter/edit.py:1]: That sounds more accurate

File: experiment_scheduler/master/master.py [hunk 4]
```diff
@@ -21,13 +21,14 @@
...
-    AllExperimentsStatus
+    AllExperimentsStatus,
+    MasterTaskStatement,
 )
 from experiment_scheduler.common import settings
 from experiment_scheduler.common.settings import USER_CONFIG
 from experiment_scheduler.common.logging import get_logger, start_end_logger
 
-
+# pylint: disable=E1101
```
[the remark this answers is not in the document]
Comment (ptaejoon) [reply experiment_scheduler/master/master.py:31]: Current Problem is pylint cannot recognise grpc class's member as it is. I suggest to exclude E1101 from pylint configuration

[the remark this answers is not in the document]
Comment (ptaejoon) [reply experiment_scheduler/master/master.py:31]: Disable comment will be deleted after precommit fixed properly

**Comment by ptaejoon** [comment]
> Thanks for your work! It's quite nice feature. Could you update README file and add unit tests?

Currently, No fully functional test codes exists. I will fill up unit test codes after settings for test code finished.
Plus, README.md also does not provide command lists.

**Review by sinunu — APPROVED** [review]

**Pull request merged**